elastic/kibana
# [ObsUX] Support all metadata fields in asset details flyout

Closes #163489 
## Summary

This PR adds support for all metadata fields to be displayed in the assets details view (flyout and page). It removes the current predefined mapping. The `host`, `agent`, and `cloud` groups will stay the same and we will show all properties present in the groups. The other metadata table functionality stays the same.

### Before

[Old Metadata Mapping](https://github.com/elastic/kibana/blob/e804ef2dc84f9518807d3ee48d878d3ba6173807/x-pack/plugins/infra/public/components/asset_details/tabs/metadata/utils.ts)

![image](https://github.com/elastic/kibana/assets/14139027/83cefdf6-f7dd-48b1-8564-bb6a17eccc37)

### After

Shows all metadata received from the API for `host`, `agent`, and `cloud` groups 

![image](https://github.com/elastic/kibana/assets/14139027/2912845f-be87-4b22-8db2-8bb6aefa9e58)

## Testing 
- Go to Host view and open the flyout for a host (selecting the metadata tab)
- Check the metadata table
    - All fields returned by the API should be visible: 
    
    <img width="1740" alt="image" src="https://github.com/elastic/kibana/assets/14139027/90d3e474-162a-4b82-af80-958db9f80522">

    - Check the pin and add/remove filter functionalities and make sure they still work (also for the "new" fields)
    - Open it as a page and do the same check
- Open inventory and click on a host - Same checks should be done there in the asset details flyout and full page
